compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp

```diff
@@ -675,15 +713,59 @@ class ConvertNCHWToNHWC final : public luci::CircleNodeMutableVisitor<bool>
       return false;
 
     auto pre_trans = create_pre_transpose(node);
-    pre_trans->a(node->input());
+    pre_trans->a(input);
     node->input(pre_trans);
 
     // Do shape inference for this node again.
     node->shape_status(luci::ShapeStatus::UNDEFINED);
 
     node->reduction_indices(nhwc_rindices);
 
-    auto post_trans = create_post_transpose(node);
+    if (node->keep_dims())
+    {
+      auto post_trans = create_post_transpose(node);
+      loco::replace(node).with(post_trans);
+
+      post_trans->a(node);
+
+      return true;
+    }
+
+    // node->keep_dims() == false
+    // 1D output never needs a transpose
+    if (node->rank() <= 1)
+      return true;
+
+    std::vector<bool> reduced_dims_nhwc(4, false);
+    uint32_t num_reduced_indices = nhwc_rindices->size<loco::DataType::S32>();
+
+    for (uint32_t ri = 0; ri < num_reduced_indices; ++ri)
+    {
+      reduced_dims_nhwc[nhwc_rindices->at<loco::DataType::S32>(ri)] = true;
+    }
+
+    // if channel dimension has been reduced, we don't need a transpose
+    if (reduced_dims_nhwc[3])
+      return true;
```

Comment:
@jinevening , Q) return true here will mark this node as `NHWC`, is this correct?

Reply:
I think it is ok.

The output is not rank 4, but the behavior fits our intention (for removing redundant Transpose).